sbi/inference/posteriors/base_posterior.py

```diff
@@ -151,19 +154,22 @@ def map(
         save_best_every: int = 10,
         show_progress_bars: bool = False,
     ) -> Tensor:
-        """See child classes for docstring."""
-        self.potential_fn.set_x(self._x_else_default_x(x))
+        """Calculates the maximum-a-posteriori estimate (MAP).
+
+        See `map()` method of child classes for docstring.
+        """
+        self.potential_fn.set_x(self.default_x)
 
         if init_method == "posterior":
-            inits = self.sample((num_init_samples,), self._x_else_default_x(x))
+            inits = self.sample((num_init_samples,))
```

Comment:
maybe add an `assert self.default_x` here to make sure it is defined and if not, to give an short error message tailored to `MAP`?

Reply:
I see, it's handled in `map`, so all good.